apps/experiments/views/experiment.py

```diff
@@ -528,19 +528,32 @@ def update_delete_channel(request, team_slug: str, experiment_id: int, channel_i
 def _start_experiment_session(
     experiment: Experiment,
     experiment_channel: ExperimentChannel,
-    user: CustomUser | None = None,
-    participant: Participant | None = None,
-    external_chat_id: str | None = None,
+    participant_user: CustomUser | None = None,
+    participant_identifier: str = "",
 ) -> ExperimentSession:
-    session = ExperimentSession.objects.create(
-        team=experiment.team,
-        user=user,
-        participant=participant,
-        experiment=experiment,
-        llm=experiment.llm,
-        external_chat_id=external_chat_id,
-        experiment_channel=experiment_channel,
-    )
+    with transaction.atomic():
+        session = ExperimentSession.objects.create(
```

Comment:
if you move this to after the participant is created then you don't need to do an additional save.